apps/src/experiments.js

```diff
@@ -16,29 +21,61 @@ experiments.getQueryString_ = function () {
   return window.location.search;
 };
 
+experiments.getEnabledExperiments = function () {
+  var jsonList = localStorage.getItem(STORAGE_KEY);
+  var enabled = jsonList ? JSON.parse(jsonList) : [];
+  OLD_KEY_WHITELIST.forEach(function (key) {
+    if (localStorage.getItem('experiments-' + key) === 'true') {
+      enabled.push(key);
+    }
+  });
+  return enabled;
+};
+
+experiments.setEnabled = function (key, shouldEnable) {
+  var allEnabled = this.getEnabledExperiments();
+  if (!allEnabled.includes(key) && shouldEnable) {
+    allEnabled.push(key);
+  } else if (allEnabled.includes(key) && !shouldEnable) {
+    allEnabled.splice(allEnabled.indexOf(key), 1);
+  } else {
+    return;
+  }
+  localStorage.setItem(STORAGE_KEY, JSON.stringify(allEnabled));
+};
+
 /**
  * Checks whether provided experiment is enabled or not
  * @param {string} key - Name of experiment in question
  * @returns {bool}
  */
 experiments.isEnabled = function (key) {
-  var enabled;
-  var queryString = this.getQueryString_();
-
-  // check query string, and update localStorage if necessary
-  var regex = new RegExp(key + '=(true|false)');
-  var match = regex.exec(queryString);
-  if (match) {
-    enabled = match[1] === 'true';
-    if (enabled) {
-      localStorage.setItem('experiments-' + key, 'true');
-    } else {
-      localStorage.removeItem('experiments-' + key);
+  var enabled = this.getEnabledExperiments().includes(key);
+  var query = queryString.parse(this.getQueryString_());
+
+  var enableQuery = query['enableExperiments'];
+  var disableQuery = query['disableExperiments'];
+  var deprecatedKeyQuery = query[key];
+
+  if (enableQuery) {
+    var experimentsToEnable = enableQuery.split(',');
+    if (experimentsToEnable.includes(key)) {
+      enabled = true;
+      this.setEnabled(key, true);
     }
-  } else {
-    // key not in query string, go look at local storage
-    enabled = (localStorage.getItem('experiments-' + key) === 'true');
   }
 
+  if (disableQuery) {
+    var experimentsToDisable = disableQuery.split(',');
+    if (experimentsToDisable.includes(key)) {
+      enabled = false;
+      this.setEnabled(key, false);
+    }
+  }
+
+  if (OLD_KEY_WHITELIST.includes(key) && deprecatedKeyQuery) {
+    enabled = deprecatedKeyQuery === 'true';
+    this.setEnabled(key, enabled);
+  }
   return enabled;
 };
```

Comment:
yep, I agree that this is confusing. I'd rather address it in a separate diff though since it's not a behavior change. We could just update local storage when the module is loaded, which would fix this issue.